File: GoLang/6 kyu/Is a number prime/main.go
package main

import (
	"fmt"
	"math"
)

func main() {
	fmt.Println(IsPrime(7))
}

func IsPrime(n int) bool {
  if n <= 1 {
    return false
  }
  if (n == 2 || n == 3) {
    return true
  }
  if (n % 2 == 0 || n % 3 == 0) {
    return false
  }
  
  for i := 5; i <= int(math.Sqrt(float64(n))); i += 6 {
    if (n % i == 0 || n % (i + 2) == 0) {
      return  false
    }
  }
  
  return true
}

File: Python/6 kyu/Is a number prime/main.py
from math import sqrt

def is_prime(num):
    if num <= 1: return False
    if num == 2 or num == 3: return True
    if num%2 == 0 or num%3 == 0: return False
    for i in range(5, int(sqrt(num) + 1), 6):
        if (num % i == 0) or (num % (i + 2) == 0): return False
    return True 

if __name__ == "__main__":
    print(is_prime(7))